### PR TITLE
complete the challenge

### PR DESCRIPTION
**Pull request by yasirelbouazzati**
i complete the all tasks of the challenge

### CODE REVIEW

File: .DS_Store
Comment (tictools) [inline .DS_Store:1]: FIX: add `.DS_Store` to `.gitignore` and remove this file

File: exercise_01/getDistanceFromThreshold.js
Comment (tictools) [inline exercise_01/getDistanceFromThreshold.js:1]: Comment was defining the name of the params (value, threshold)

```js
/**
* #1 :: ...
 * input: number (value) 👈
 * input: number (threshold) 👈
 * output: number
 */
```
FIX: rename params

File: exercise_01/index.js
Comment (tictools) [inline exercise_01/index.js:1]: FIX: rename argument `value_2`

File: exercise_02/getTotalWordsFromString.js [hunk 1]
```diff
@@ -3,3 +3,16 @@
...
+
+/**
+ * @param {string}  sentence
+ * @returns {number}
+ */
+
+export default function getTotalWordsFromString(sentence) {
+  const words = sentence.split(/\s+/);
+
+  const totalWords = words.filter((word) => word !== "").length;
```
Comment (tictools) [inline exercise_02/getTotalWordsFromString.js:15]: Do you understand what you are doing here? How does `filter` work?... 🤔
We haven't still worked with array methods.

**Review by tictools — CHANGES_REQUESTED** [review]

File: index.html
Comment (AGUACATE2304) [inline index.html:1]: Project Requirements didn't asked to create this file.

File: exercise_03/generateRandomNumberInRange.js
Comment (AGUACATE2304) [inline exercise_03/generateRandomNumberInRange.js:1]: IMHO You should changes this.
const getTotalWordsFromString = (limit) {
 "Rest of code"
}
export default generateRandomNumberInRange;

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (AGUACATE2304) [inline .gitignore]: Add blank line at end of the file.

File: .prettierignore
Comment (AGUACATE2304) [inline .prettierignore:1]: Add blank line at end of the file.

File: exercise_04/isGreaterThan.js
Comment (AGUACATE2304) [inline exercise_04/isGreaterThan.js:1]: IMHO You should changes this.
const isGreaterThan = (value, threshold) {
"Rest of code"
}
export default isGreaterThan;

File: exercise_05/index.js [hunk 3]
```diff
@@ -1 +1,13 @@
 //import 'isAdditionGreaterThanFifty' and execute
+
+import isAdditionGreaterThanFifty from "./isAdditionGreaterThanFifty.js";
+
+const number1 = 30;
```
Comment (AGUACATE2304) [inline exercise_05/index.js]: Rename variable name  firstNumber

File: exercise_05/index.js [hunk 4]
```diff
@@ -1 +1,13 @@
...
+
+const number2 = 25;
```
Comment (AGUACATE2304) [inline exercise_05/index.js]: Rename variable name secondNumber

Use more concret varaible names

File: exercise_07/isTypeOf.js [hunk 5]
```diff
@@ -6,3 +6,18 @@
...
+/**
+ * @param {string | number | boolean | array}  value
+ * @param {boolean}  type
+ * @returns {boolean}
+ */
+
+export default function isTypeOf(value, type) {
+  if (type) {
+    return Array.isArray(value);
+  } else {
```
Comment (AGUACATE2304) [inline exercise_07/isTypeOf.js:20]: Did you used chalistenics.

Don't do else

File: package.json [hunk 6]
```diff
@@ -9,5 +9,9 @@
   },
```
Comment (AGUACATE2304) [inline package.json:11]: Did you create GitHub Actions??

Add script:
"lint:fix": "npx prettier . --write"

**Review by AGUACATE2304 — CHANGES_REQUESTED** [review]
You need to change repo name.

05_JavaScript_functions    >    Javascript-functions

**Review by tictools — APPROVED** [review]

**Review by AGUACATE2304 — APPROVED** [review]

**Pull request merged**